base_layer/core/src/mining/miner.rs

```diff
@@ -109,17 +109,19 @@ impl<B: BlockchainBackend> Miner<B> {
         while !self.kill_flag.load(Ordering::Relaxed) {
             while !self.received_new_block_flag.load(Ordering::Relaxed) {
```

Comment:
If this is initially false, you will never mine a block...

Reply:
I thought the same thing, this flag must change to true when a block is assembled @SWvheerden ?

Reply:
It should not start mining blocks when the miner is started. The miner is started before the base node is ready to give it blocks. So the miner waits until the basenode is in listing mode or received a new block. Look below at the code that listens to the basenode events. The flag is swapped to true there.

Reply:
The issue was that I had peers set in the whitelist that were unreachable, so the node never went into `Listening` mode; so the flag never got set.